src/api/publish.rs

```diff
@@ -1,8 +1,16 @@
-use rocket::serde::Deserialize;
+use rocket::serde::{Deserialize, Serialize};
+use uuid::Uuid;
 
 /// The publish request.
 #[derive(Deserialize, Debug)]
 pub struct PublishRequest {
     pub name: String,
     pub version: String,
+    pub upload_id: String,
+}
+
+/// The response to an upload_project request.
+#[derive(Serialize, Debug)]
+pub struct UploadResponse {
```

Comment:
nit?
```suggestion
pub struct PublishResponse {
```

Reply:
This is actually the response to the `upload_project` API. I put in in the same file as publish since it's part of the larger publishing workflow but I could move it to a separate file if this is confusing.